### PR TITLE
vala.eclass: version 0.24 has been dropped from the tree

### PR DESCRIPTION
**Pull request by dave-kennedy**
Fixes bug 583922.

Reported-by: Dave Kennedy <dave-kennedy@outlook.com>

### CODE REVIEW

File: eclass/vala.eclass [hunk 1]
```diff
@@ -50,13 +50,16 @@ vala_api_versions() {
...
+	minimal_supported_minor_version="26"
 
 	for ((minor_version = ${VALA_MAX_API_VERSION#*.}; minor_version >= ${VALA_MIN_API_VERSION#*.}; minor_version = minor_version - 2)); do
 		if ((minor_version >= minimal_supported_minor_version)); then
 			echo "0.${minor_version}"
 		fi
 	done
+
+	# For some reason 0.22 is still in the tree but 0.24 is not
+	echo "0.22"
```
Comment (mgorny) [inline eclass/vala.eclass]: I'm pretty sure this is wrong. The description says 'from VALA_MAX_API_VERSION down to VALA_MIN_API_VERSION'.

Comment (dave-kennedy) [inline eclass/vala.eclass]: You mean 0.24 should not have been removed from the tree?

Comment (dave-kennedy) [inline eclass/vala.eclass]: Can we change the description?

Comment (mgorny) [inline eclass/vala.eclass]: What's the problem having just 0.26+ there?

Comment (dave-kennedy) [inline eclass/vala.eclass]: Fine with me, but I don't know the backstory. Who would?

**Comment by mgorny** [comment]
@gentoo/gnome

**Comment by pacho2** [comment]
We needed 0.22 for lxsession, I was pushing a lot for its stabilization but I don't know if it was finished by exotic arches finally. After that, it's only a matter of killing the old vala version and update the eclass... hence, this "hacks" won't be accepted as-is :/

**Comment by dave-kennedy** [comment]
@pacho2 the script is currently broken. What do you propose?

**Comment by pacho2** [comment]
It only causes a minor warning on repoman until we kill the old version... if you know how to make the snipped in https://bugs.gentoo.org/show_bug.cgi?id=583922 to not output deps for non-existing ebuilds... feel free to show it there. But adding this hardcoding is not the way to go

**Comment by pacho2** [comment]
And, for this concrete case, we simply need some time to drop the old vala version once we check all reverse deps are finally dropped (it should be lxsession only... but it's better to double check before killing 0.22). Then, we can raise the minimum to a newer version in eclass.

**Pull request closed without merging**